runtime/compiler/vimdoc.vim

```diff
@@ -0,0 +1,17 @@
+" Vim Compiler File
+" Language:             vimdoc
+" Maintainer:           Wu, Zhenyu <wuzhenyu@ustc.edu>
+" Latest Revision:      2024-04-09
+
+if exists('b:current_compiler')
+  finish
+endif
+let b:current_compiler = 'vimdoc'
+
+let s:save_cpoptions = &cpoptions
+set cpoptions&vim
+
+CompilerSet makeprg=vimdoc\ .
```

Comment:
Can you add a comment where to get vimdoc?

Reply:
> vimdoc\ .

I suppose this is should run vimdoc with the currently edited directory? `.` might be different for that however because of `:cd/:tcd` etc. 

Also, does this work correctly, if there isn't a `doc/` directory?

Reply:
> Also, does this work correctly, if there isn't a doc/ directory?

It will `mkdir doc`.

> run vimdoc with the currently edited directory

In my case, if my `$PWD` is `/the/directory/of/a/vim/plugin` and I am editing `/the/directory/of/a/vim/plugin/autoload/my_plugin/my_functions.vim` or `/the/directory/of/a/vim/plugin/plugin/my_plugin.vim`, Run `vimdoc .` will work.

Reply:
More details about vimdoc you can ask the maintainer of vimdoc @dbarnett

Reply:
Yeah, but what if it is  `/tmp` ? I guess `:makeprg` should rather be `makeprg=vimdoc\ %:h:S` (similar to what other compiler plugins set).

Reply:
>  makeprg=vimdoc\ %:h:S

Oh, we can search `addon.json` to know the path `XXX` of `vimdoc XXX`

https://github.com/google/vim-maktaba/tree/HEAD/examples/helloworld

Reply:
not sure what `addon.json` is supposed to be doing and the URL doesn't explain it. I have never seen this with vimscripts so if we start searching for that file, it probably makes the compiler not useful anymore...

Reply:
https://github.com/google/vim-maktaba/blob/fe95bb10f6bb250943a44632107f6a3d76ce5f28/examples/helloworld/addon-info.json

Sorry, should be addon-info.json

Reply:
I think upstream should have a default path of `vimdoc PATH` :smile:

Reply:
so let's just remove the directory from the :make command as @dkearns mentions. Let's not overcomplicate things then.

Reply:
Fixed.